web/src/features/weather-layers/wind-layer/calc.ts

```diff
@@ -0,0 +1,91 @@
+import { MapboxMap } from 'react-map-gl';
+// u, v, m
+export type WindVector = [number, number, number];
+
+/**
+ * interpolation for vectors like wind (u,v,m)
+ * @returns {WindVector}
+ */
+export const bilinearInterpolateVector = (
+  x: number,
+  y: number,
+  g00: number[],
+  g10: number[],
+  g01: number[],
+  g11: number[]
+): WindVector => {
+  const rx = 1 - x;
+  const ry = 1 - y;
+  const a = rx * ry,
+    b = x * ry,
+    c = rx * y,
+    d = x * y;
+  const u = g00[0] * a + g10[0] * b + g01[0] * c + g11[0] * d;
+  const v = g00[1] * a + g10[1] * b + g01[1] * c + g11[1] * d;
+  return [u, v, Math.hypot(u, v)];
+};
```

Comment:
It would be great if we could add some unit tests to these, there are a bunch of math going on and while I think they look correct it would be nice to have a test in place for it.